client/app/components/directions.jsx

```diff
@@ -20,9 +22,22 @@ class Directions extends React.Component {
   }
 
   componentDidUpdate() {
-    if (this.props.origDest) {
+    if (this.props.map) {
+      this.state.directionsDisplay.setMap(this.props.map);
+    }
+    if (this.props.origDest && this.props.origDest !== this.state.origDest) {
       this.calcRoute(this.props.origDest[0].formatted_address, this.props.origDest[1].formatted_address);
+      this.setState({
```

Comment:
Question. It looks like you are setting state within ComponentDidUpdate. Won't this create an infinite loop, or is there something I'm missing?